components/script/dom/node.rs

```diff
@@ -604,6 +604,10 @@ impl Node {
         }
     }
 
+    pub fn is_empty(&self) -> bool {
+        self.data.len() == 0
```

Comment:
`self.data` doesn't even exist??

You will need a `match self.type_id()` and do something analogous to `len()`, but without calling `CharacterData.Length()`

Reply:
I would revert this change for now and might solve it separately